packages/datadog-plugin-jest/src/jest-jasmine2.js

```diff
@@ -0,0 +1,199 @@
+  return function wrapIt (it) {
+    return function itWithTrace (description, specFunction, timeout) {
+      let oldSpecFunction = specFunction
+      if (specFunction.length) {
+        oldSpecFunction = promisify(oldSpecFunction)
+      }
+
+      const { childOf, commonSpanTags } = getTestSpanTags(tracer, testEnvironmentMetadata)
+
+      const testSuite = globalInput.jasmine.testPath.replace(`${globalConfig.rootDir}/`, '')
+
+      const newSpecFunction = tracer.wrap(
+        'jest.test',
+        {
+          type: 'test',
+          childOf,
+          tags: { ...commonSpanTags, [TEST_SUITE]: testSuite }
+        },
+        async (done) => {
+          const testSpan = tracer.scope().active()
+          const { currentTestName } = globalInput.expect.getState()
+          const resource = `${testSuite}.${currentTestName}`
+          testSpan.setTag(TEST_NAME, currentTestName)
+          testSpan.setTag(RESOURCE_NAME, resource)
+          testSpan.context()._trace.origin = CI_APP_ORIGIN
+          let result
+          globalInput.jasmine.testSpanByTestName[currentTestName] = testSpan
+
+          try {
+            result = await oldSpecFunction()
+            const suppressedErrors = globalInput.expect.getState().suppressedErrors
+            if (suppressedErrors && suppressedErrors.length) {
+              testSpan.setTag('error', suppressedErrors[0])
+              testSpan.setTag(TEST_STATUS, 'fail')
+            }
+            if (!testSpan._spanContext._tags[TEST_STATUS]) {
+              testSpan.setTag(TEST_STATUS, 'pass')
+            }
+          } catch (error) {
+            testSpan.setTag(TEST_STATUS, 'fail')
+            testSpan.setTag('error', error)
+            if (done) {
+              done(error)
+            }
+            throw error
+          } finally {
+            finishAllTraceSpans(testSpan)
+          }
+          if (done) {
+            done(result)
+          }
+        }
+      )
+      return it(description, newSpecFunction, timeout)
+    }
+  }
+}
+
+function createWrapOnException (tracer, globalInput) {
+  return function wrapOnException (onException) {
+    return function onExceptionWithTrace (err) {
+      let activeTestSpan = tracer.scope().active()
+      if (!activeTestSpan) {
+        activeTestSpan = globalInput.jasmine.testSpanByTestName[this.getFullName()]
+      }
+      if (!activeTestSpan) {
+        return onException.apply(this, arguments)
+      }
+      const {
+        [TEST_NAME]: testName,
+        [TEST_SUITE]: testSuite,
+        [TEST_STATUS]: testStatus
+      } = activeTestSpan._spanContext._tags
+
+      const isActiveSpanFailing = this.getFullName() === testName &&
+        this.result.testPath.endsWith(testSuite)
+
+      if (isActiveSpanFailing && !testStatus) {
+        activeTestSpan.setTag(TEST_STATUS, 'fail')
+        // If we don't do this, jest will show this file on its error message
```

Comment:
In the message itself on only in the stack? Right now we don't try to avoid being in the stack anywhere else, so that would be a larger discussion.

Reply:
in the message itself. If it was just the stack it'd be fine